src/reader/torch_data_loader_reader.py

```diff
@@ -97,9 +97,10 @@ def read(self, epoch_number):
         # superclass function shuffle the file list 
         super().read(epoch_number)
         do_shuffle = True if self.memory_shuffle != Shuffle.OFF else False
-        
-        dataset = TorchDataset(self._file_list, self.my_rank, self.format)
-        
+        if self._args.num_samples_per_file == 1:
+            dataset = TorchDataset(self._file_list, self.my_rank, self.format)
```

Comment:
We should create a issue to support multiple files for PyTorch so that we can pick it up later.